src/test/java/com/devonfw/sample/archunit/AvoidCyclicDependenciesTest.java

```diff
@@ -0,0 +1,23 @@
+package com.devonfw.sample.archunit;
+
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+
+import static com.tngtech.archunit.library.dependencies.SlicesRuleDefinition.slices;
+
+import com.tngtech.archunit.core.importer.ImportOption;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class AvoidCyclicDependenciesTest {
+    @ArchTest
+    static final ArchRule no_cyclic_dependencies_are_allowed =
+    slices()
+        .matching("..archunit.(*)..")
```

Comment:
here we encode our project specific namespace into the rule.
The first thing that projects will have to do after refactoring this sample to their own project namespace is to identify all these places and change them according to their project.
With `StructuredPackage` we have created a solution that is fully generic and should apply to all devonfw based Java projects no matter what their package namespace actually is.
I do not know how this ArchUnit patterns are working. Are they Java regular expressions?
As `..` behaves differently from what I know about regex this looks more special.
However, I would simply try something like this instead:
```
  .matching("..(*).(common|service|logic|dataaccess|batch|client")..")
```

Reply:
Your approach does work. I have adjusted it in the file and we will also adjust it in #29 as well.